packages/contentstack-import/src/utils/entries-helper.ts

```diff
@@ -201,20 +203,51 @@ export const lookupEntries = function (
 
   uids = _.uniq(uids);
   let entry = JSON.stringify(data.entry);
+
+  //Note: Use this in case prod breaks
+  // uids?.forEach(function (uid: any) {
+  //   if (mappedUids.hasOwnProperty(uid)) {
+  //     const sanitizedUid = escapeRegExp(uid);
+  //     const escapedMappedUid = escapeRegExp(mappedUids[uid]);
+  //     const uidRegex = new RegExp(`\\b${sanitizedUid}\\b`, 'img');
+  //     let { status } = validateRegex(uidRegex);
+  //     if (status === 'safe') {
+  //       entry = entry.replace(uidRegex, escapedMappedUid);
+  //       mapped.push(uid);
+  //     }
+  //   } else {
+  //     unmapped.push(uid);
+  //   }
+  // });
+  
   uids?.forEach(function (uid: any) {
     if (mappedUids.hasOwnProperty(uid)) {
       const sanitizedUid = escapeRegExp(uid);
       const escapedMappedUid = escapeRegExp(mappedUids[uid]);
-      const uidRegex = new RegExp(`\\b${sanitizedUid}\\b`, 'img');
-      let { status } = validateRegex(uidRegex);
-      if (status === 'safe') {
-        entry = entry.replace(uidRegex, escapedMappedUid);
-        mapped.push(uid);
+      let index = entry.indexOf(sanitizedUid);
+      while (index !== -1) {
+        if ((index === 0 || !isWordCharacter(entry.charAt(index - 1))) &&
+            (index + sanitizedUid.length === entry.length || !isWordCharacter(entry.charAt(index + sanitizedUid.length)))) {
+          entry = entry.substring(0, index) + escapedMappedUid + entry.substring(index + sanitizedUid.length);
+          index += escapedMappedUid.length;  
+        } else {
+          index += sanitizedUid.length;  
+        }
+        index = entry.indexOf(sanitizedUid, index); 
       }
+      mapped.push(uid);
```

Comment:
@cs-raj , I think this code is more complex and harder to read compared to the regex approach. Might be it'll create problem or less efficient in case of larger strings.
This word char can be error prone.

Reply:
@cs-raj Why splitting logic used above is not possible here ?
@cs-raj Why we have to remove the regex implementation, what exactly error here ?

Reply:
Removed and replaced with older code